### PR TITLE
Morecast 2 styling summary

### PR DESCRIPTION
**Pull request by conbrad**
- Styling for summary page
    - Style defined in `Root` wrapper around respective `DataGrid`s and referenced using the cell class name handlers provided by `DataGrid`
- Reuses forecast dropdown across summary headers
    - Replaces `ApplyToColumnMenu` component with `ForecastHeader` component and passes down clicked column definition handlers to each column
# Test Links:
[Landing Page](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3468.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Comment by dgboss** [comment]
Closing to save resources.

File: web/src/features/moreCast2/components/forecastHeader.test.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  it('should render an enabled forecast button without menu enabled by default', () => {
+    const { getByTestId, queryByTestId } = renderForecastHeader()
+
+    const forecastButton = getByTestId('testForecast-column-header')
+    expect(forecastButton).toBeInTheDocument()
+    expect(forecastButton).toBeEnabled()
+    expect(queryByTestId('apply-to-column-menu')).not.toBeInTheDocument()
+  })
+  it('should open apply to column menu when forecast button clicked', async () => {
```
Comment (dgboss) [inline web/src/features/moreCast2/components/forecastHeader.test.tsx:40]: Is this test complete?

File: web/src/app/theme.ts [hunk 2]
```diff
@@ -120,22 +134,9 @@ export const modelColorClass = (params: Pick<GridCellParams | GridColumnHeaderPa
...
-  if (params.field.includes('NAM')) {
-    return 'nam'
-  }
-  if (params.field.includes('RDPS')) {
-    return 'rdps'
-  }
-  return ''
+  const stringKeys = Object.keys(MORECAST_MODEL_COLORS)
+  const modelKey = stringKeys.find(key => params.field.includes(key.toUpperCase()))
+  return modelKey ? modelKey : ''
```
Comment (dgboss) [inline web/src/app/theme.ts:139]: SonarCloud doesn't like this.

**Review by dgboss — APPROVED** [review]
Looking good! Just a couple nit picky things.

File: web/src/features/moreCast2/components/ForecastSummaryDataGrid.tsx [hunk 3]
```diff
@@ -1,52 +1,67 @@
...
+
+  MORECAST2_INDEX_FIELDS.forEach(indexField => {
+    styles[`& .${indexField.getField()}-forecast-header`] = {
+      backgroundColor: `rgba(0, 51, 102, 1)`,
+      color: theme.palette.common.white
+    }
+  })
+
+  styles[`& .forecastCell`] = {
+    backgroundColor: 'rgba(208, 208, 208, 1)'
```
Comment (brettedw) [inline web/src/features/moreCast2/components/ForecastSummaryDataGrid.tsx:47]: I think the colour here might actually be `rgba(238,238,238,1)` in the figma design. I had a hard time clicking through to get that, it just looked darker than the design to me

File: web/src/features/moreCast2/components/forecastHeader.test.tsx (hunk 1 above)
Comment (conbrad) [reply web/src/features/moreCast2/components/forecastHeader.test.tsx:40]: Yep, it'll fail if the function it calls fails -- it's reused code for the next test

File: web/src/features/moreCast2/components/ForecastCell.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { Grid, TextField, Tooltip } from '@mui/material'
 import { GridRenderCellParams } from '@mui/x-data-grid'
 import RemoveCircleIcon from '@mui/icons-material/RemoveCircle'
 import AddBoxIcon from '@mui/icons-material/AddBox'
-import { MEDIUM_GREY } from 'app/theme'
+import { MEDIUM_GREY, theme } from 'app/theme'
```
[the remark this answers is not in the document]
Comment (conbrad) [reply web/src/features/moreCast2/components/ForecastCell.tsx:6]: Not sure either, we could make a dev task to change it all over

File: web/src/features/moreCast2/components/ForecastDataGrid.tsx [hunk 5]
```diff
@@ -47,52 +46,50 @@ const Root = styled('div')(() => {
...
       borderRight: 'solid',
+      borderWidth: '1px',
+
       // Ugly override, tried to avoid, but MUI overwrites border with it's own otherwise
       borderColor: `${MORECAST_MODEL_COLORS[key as keyof MoreCastModelColors].border} !important`
     }
   })
 
+  styles[`& .forecastCell`] = {
+    backgroundColor: 'rgba(208, 208, 208, 1)'
```
Comment (brettedw) [inline web/src/features/moreCast2/components/ForecastDataGrid.tsx:65]: Sorry, here too

File: web/src/features/moreCast2/components/forecastHeader.test.tsx (hunk 1 above)
Comment (dgboss) [reply web/src/features/moreCast2/components/forecastHeader.test.tsx:40]: Ah, I see now!

**Review by brettedw — APPROVED** [review]
Nice work!

**Pull request merged**